integration_tests/DATA_GEN.md

```diff
@@ -0,0 +1,402 @@
+|             agg_key| min(value)|max(value)|         sum(value)|
++--------------------+-----------+----------+-------------------+
+|-5395130417820713410|-2146262091|2137753315|-790927616093345360|
+|-2676343845813715061|-2146262091|2137753315|-790805185346631680|
+|-2519888240660072622|-2146262091|2137753315|-790643842502125280|
+| -423375629809417248|-2146262091|2137753315|-790620036523597620|
+| 2201036318770978408|-2146262091|2137753315|-790864265299987700|
+| 2236810499658231116|-2146262091|2137753315|-791190430932705540|
+| 5462872210444757997|-2146262091|2137753315|-791094653391187280|
+| 5615780013263136643|-2146262091|2137753315|-790580412619137960|
+| 6937819975370740711|-2146262091|2137753315|-791081287243774740|
+| 8365571621216757512|-2146262091|2137753315|-791155236047506840|
++--------------------+-----------+----------+-------------------+
+
+Time taken: 890163 ms
+```
+
+Or you could run it with the RAPIDS Accelerator where the data generations is still
+done on the CPU and cut the time down to just 75,696 ms
+
+But what if we wanted to join on multiple columns, or do an aggregation with
+multiple columns as the keys. Or what about nested columns where there are structs
+or arrays in the key set. This gets to be much more difficult. To help with this
+we provide key groups. A key group is set on a table using the `configureKeyGroup`
+API. There are currently two types of key groups supported.
+
+### CorrelatedKeyGroup
+
+A correlated key group is a group of columns and child columns where the code to
+generate the seed is normalized so that for each row the same seed is passed into
+all the generator functions. (This is not 100% correct for arrays and maps, but it
+is close enough). This results in the generated data being correlated with each
+other so that if you set a seed range of 1 to 200, you will get 200 unique values
+in each column, and 200 unique values for any combination of the keys in that 
+key group.
+
+This should work with any distribution and any type you want. The key to making
+this work is that you need to configure the value ranges the same for both sets
+of corresponding keys. In most cases you want the types to be the same as well, 
+but Spark supports equi-joins where the left and right keys are different types.
+The default generators for integral types should produce the same values for the
+same input keys if the value range is the same for both. This is not true for
+floating point (no one should ever join on floating point values), or decimal
+types, unless the scale is the same for both sides.
+
+```scala
+val dataTable = DBGen().addTable("data", "a byte, b long", 3)
+dataTable.toDF(spark).show()
++----+--------------------+
+|   a|                   b|
++----+--------------------+
+|-120| 3448826316550928693|
+|  26|-1617384775774370579|
+| -43|-2697761218773508623|
++----+--------------------+
+dataTable.configureKeyGroup(Seq("a", "b"), CorrelatedKeyGroup(1, 0, 1), FlatDistribution())
```

Comment:
Maybe a hint about what the params to `CorrelatedKeyGroup` mean will avoid a user looking elsewhere for an answer?